File: CarAgency/ClientApp/src/models/finanzas/Transaccion.ts
import {ITransaccion} from 'models/interfaces/finanzas/ITransaccion';
import ConceptoFinanciero from './ConceptoFinanciero';
import Cuenta from './Cuenta';
import TipoOperacion from './TipoOperacion';

export class Transaccion implements ITransaccion{
    constructor(){
      this.TransaccionId = undefined;
      this.Concepto= undefined;
      this.TipoOperacion= undefined;
      this.Origen= undefined;
      this.Destino= undefined;
      this.Monto = 0;
      this.CreateDateTime = new Date();
      this.CreateUser = "";
      this.UpdateDateTime = undefined;
      this.UpdateUser = "";
      this.DeleteDateTime = undefined;
      this.DeleteUser = "";
      this.Active= true;
    };
  
    TransaccionId?: number | undefined;
    Concepto?: ConceptoFinanciero;
    TipoOperacion?: TipoOperacion;
    Origen?: Cuenta;
    Destino?: Cuenta;
    Monto: number;
    CreateDateTime?: Date | undefined;
    CreateUser: string;
    UpdateDateTime?: Date | undefined;
    UpdateUser: string;
    DeleteDateTime?: Date | undefined;
    DeleteUser?: string | undefined;
    Active: Boolean;
    
  }

File: Domain/Models/Finanzas/Transaccion.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace Domain.Models.Finanzas
{
    public class Transaccion : ClaseBase
    {
        [Key]
        public int TransaccionId { get; set; }

        public ConceptoFinanciero Concepto { get; set;}

        public TipoOperacion TipoOperacion { get; set; }

        public Cuenta Origen { get; set; }

        public Cuenta Destino { get; set; }

        [Column(TypeName = "Decimal(18,2)")]
        public decimal Monto { get; set; }

        public int? VehiculoId { get; set; }

    }
}
